internal/dag/cache_test.go

```diff
@@ -361,6 +363,60 @@ func TestKubernetesCacheInsert(t *testing.T) {
 			},
 			want: true,
 		},
+		"insert certificate configmap not referenced": {
+			obj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ca",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					CACertificateKey: fixture.CERTIFICATE,
+				},
+			},
+			want: false,
+		},
+		"insert generic configmap not referenced": {
+			obj: &v1.ConfigMap{
+				ObjectMeta: metav1.ObjectMeta{
+					Name:      "ca",
+					Namespace: "default",
+				},
+				Data: map[string]string{
+					"not-ca.crt": fixture.CERTIFICATE,
+				},
+			},
+			want: false,
+		},
+		"insert certificate configmap referenced by BackendTLSPolicy": {
```

Comment:
should be able to add a case for a Secret referenced by a BackendTLSPolicy (on the remove case as well)

Reply:
Done. It should be noted that the secretRetriggerBuild function always returns true on ca certs, which is what is always provided to backendtlspolicies, so no changes to the implementation was done here, but testing it is still valuable just to provide intent in case the logic here changes.